src/exx_kernel_default.f90

```diff
@@ -1156,14 +1160,17 @@ subroutine m_kern_exx_cri(k_off, kpart, ib_nd_acc, ibaddr, nbnab, &
                    call exx_phi_on_grid(inode,jb%global_num,jb%spec,extent, &
                         jb%xyz,jb%nsup,phi_j,r_int,xyz_zero)             
                    !
-                   if ( exx_alloc ) call exx_mem_alloc(extent,0,0,'Ome_kj','alloc')
+                   if ( exx_alloc ) call exx_mem_alloc(extent,0,0,'Ome_kj_1d_buffer','alloc')
                    !
                    call start_timer(tmr_std_exx_accumul)
-                   !$omp parallel do schedule(runtime) collapse(2) default(none) reduction(+: c) &
+                   !$omp parallel default(none) reduction(+: c) &
                    !$omp     shared(kg,jb,tmr_std_exx_poisson,tmr_std_exx_accumul,Phy_k,phi_j,phi_k,ncbeg,ia,tmr_std_exx_matmult,ewald_pot,phi_i, &
                    !$omp            exx_psolver,exx_pscheme,extent,dv,ewald_rho,inode,pulay_radius,p_omega,p_gauss,w_gauss,reckernel_3d,r_int) &
-                   !$omp     private(nsf1,nsf2,work_out_3d,work_in_3d,ewald_charge,Ome_kj, &
-                   !$omp             ncaddr,nsf3,exx_mat_elem,r,s,t)
+                   !$omp     private(nsf1,nsf2,work_out_3d,work_in_3d,ewald_charge,Ome_kj_1d_buffer, &
+                   !$omp             ncaddr,nsf3,exx_mat_elem,r,s,t) &
+                   !$omp     firstprivate(Ome_kj)
+                   Ome_kj(1:2*extent+1, 1:2*extent+1, 1:2*extent+1) => Ome_kj_1d_buffer

```

Comment:
Why is `Ome_kj` firstprivate here? 

When we enter the parallel region, `Ome_kj_1d_buffer` is made private and `Ome_kj` is an unassociated pointer. Does something break if `Ome_kj` is private? I can see that you want to point `Ome_kj` to the buffer after the private copies have been created, but before you start the parallel loop and I think that is the right way to do it.